test/src/e2e_vm_tests/mod.rs

```diff
@@ -288,20 +279,51 @@ pub fn run(filter_regex: Option<regex::Regex>) {
         ),
     ];
 
-    let mut number_of_tests_run = positive_project_names.iter().fold(0, |acc, (name, res)| {
-        if filter(name) {
-            assert_eq!(crate::e2e_vm_tests::harness::runs_in_vm(name), *res);
-            // cannot use partial eq on type `anyhow::Error` so I've used `matches!` here instead.
-            // https://users.rust-lang.org/t/issues-in-asserting-result/61198/3 for reference.
-            assert!(matches!(
-                crate::e2e_vm_tests::harness::test_json_abi(name),
-                Ok(())
-            ));
-            acc + 1
-        } else {
-            acc
-        }
-    });
+    let mut number_of_tests_run =
+        positive_project_names_no_abi
+            .iter()
+            .fold(0, |acc, (name, res)| {
+                if filter(name) {
+                    assert_eq!(crate::e2e_vm_tests::harness::runs_in_vm(name), *res);
+                    acc + 1
+                } else {
+                    acc
+                }
+            });
+
+    // Programs that should successfully compile, include abi and terminate
+    // with some known state. Note that if a non-contract is included
+    // it will be rejected during assertion. Please move it to
+    // `positive_project_names_no_abi` above.
+    let positive_project_names_with_abi = vec![
+        // contracts revert because this test runs them against the VM
+        // and no selectors will match
+        (
+            "should_pass/test_contracts/contract_abi_impl",
+            ProgramState::Revert(0),
+        ),
+        (
+            "should_pass/language/valid_impurity",
+            ProgramState::Revert(0), // false
+        ),
+    ];
+
+    number_of_tests_run += positive_project_names_with_abi
+        .iter()
+        .fold(0, |acc, (name, res)| {
+            if filter(name) {
+                assert_eq!(crate::e2e_vm_tests::harness::runs_in_vm(name), *res);
+                // cannot use partial eq on type `anyhow::Error` so I've used `matches!` here instead.
+                // https://users.rust-lang.org/t/issues-in-asserting-result/61198/3 for reference.
+                assert!(matches!(
+                    crate::e2e_vm_tests::harness::test_json_abi(name),
+                    Ok(())
+                ));
```

Comment:
```suggestion
                assert!
                    crate::e2e_vm_tests::harness::test_json_abi(name).is_ok()
                );
```

We can't use this because it is nightly, but just in case you were curious, there's [assert_match](https://doc.rust-lang.org/std/assert_matches/macro.assert_matches.html) coming soon (TM).

Reply:
I did see that as an option, really looking forward to some of the nightly options!

Reply:
BTW, my suggestion is still valid on stable. It doesn't use `assert_matches`:
```rust
        assert!
                    crate::e2e_vm_tests::harness::test_json_abi(name).is_ok()
                );
```